test/test_ray_elastic.py

```diff
@@ -0,0 +1,297 @@
+"""Ray-Horovod Elastic training unit tests.
```

Comment:
We should add this as a standalone test in `gen-pipeline.sh`.  I am working on a refactor so we won't need to do this for every test that doesn't get executed by `horovodrun` (#2264), but for now we need to follow the same process that we used in the last Ray PR.

Reply:
Done. Take a look, I think I did it right? (but not sure).

Reply:
Looks good!  Only additional step is to run this from the root of the repo:

```
BUILDKITE_PIPELINE_SLUG=SLUG BUILDKITE_BRANCH=BRANCH .buildkite/gen-pipeline.sh > test/data/expected_buildkite_pipeline.yaml
```

We do this to make sure that the generated pipeline is not broken.

Reply:
It printed `WARNING:root:no commit (None) or default branch (None) given` - should be OK right?

Reply:
Yeah, that's fine.